Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx

```diff
@@ -40,40 +36,29 @@ const ExternalAdapterSettings = (props: Props) => {
   const { definitions: schemaDefinitions } = schemas?.sdk?.content ?? {};
   const uiSchemas = schemas?.ui?.content ?? {};
 
-  const [currentModalProps, setModalProps] = useState<
-    { key: string; packageName: string; firstTime: boolean } | undefined
-  >();
+  const [currentModalProps, setModalProps] = useState<{ key: string; packageName: string } | undefined>();
 
-  const openModal = (key?: string, firstTime?: boolean, packageName?: string) => {
-    if (key == null || packageName == null || firstTime == null) {
+  const openModal = (key?: string, packageName?: string) => {
+    if (key == null || packageName == null) {
       setModalProps(undefined);
     } else {
-      setModalProps({ key, packageName, firstTime });
+      setModalProps({ key, packageName });
     }
   };
 
   if (schemaDefinitions == null) return null;
 
-  const externalServices = (schemas: (JSONSchema7 & { key: string; packageName?: string; firstTime?: boolean })[]) => (
+  const externalServices = (schemas: (JSONSchema7 & { key: string; packageName?: string })[]) => (
```

Comment:
There are few places this type is being used { key: string; packageName?: string }. Maybe create a type to be used across files.

In addition it looks like key always exist as a string and packageName could be undefined. There are places where it is indicated key and packageName could be optional. Could we make sure this type is consistent?